dataservice/extensions/flask_indexd.py

```diff
@@ -161,17 +162,55 @@ def update(self, record):
             "metadata": record._metadata
         }
 
-        # Update the file on indexd
+        if req_body['size'] == old['size']:
+            del req_body['size']
+        if req_body['hashes'] == old['hashes']:
+            del req_body['hashes']
+
+        # If acl changed, update all previous version with new acl
+        if record.acl != old['acl']:
+            self._update_all_acls(record)
+
         url = '{}{}?rev={}'.format(self.url, record.latest_did, record.rev)
-        resp = self.session.post(url, json=req_body)
+        if 'size' in req_body or 'hashes' in req_body:
+            # Create a new version in indxed
+            req_body['form'] = 'object'
+            resp = self.session.post(url, json=req_body)
+            did = resp.json()['did']
+            record.latest_did = did
+        else:
+            # Update the file on indexd
+            resp = self.session.put(url, json=req_body)
+
         self.check_response(resp)
         resp.raise_for_status()
 
-        did = resp.json()['did']
-        record.latest_did = did
-
         return record
 
+    def _update_all_acls(self, record):
```

Comment:
I can't think of a good case for when something might go wrong with one of the PUTs, but if any one of the PUTs came back with a non 200 status code should we check for that and abort the update? So that we can avoid mismatching acls across versions

Reply:
You'd still have mismatching acls if you updated doc A and doc B, but failed on doc C.
If you fail on doc C, you'd have to re-put doc A and doc B, and what if you fail when updating doc B etc etc ad absurdum.
This will need to be a check in our reports or indexd cleanup scripts.

Reply:
yea true..